main/src/ca/uwaterloo/flix/language/phase/Typer.scala

```diff
@@ -467,6 +467,18 @@ object Typer extends Phase[ResolvedAst.Program, TypedAst.Root] {
         */
       def visitExp(e0: ResolvedAst.Expression): InferMonad[Type] = e0 match {
 
+        /*
+         * Statement expression.
+         */
+        case ResolvedAst.Expression.Statement(exp1, exp2, tvar, loc) =>
+          // exp1 : t1    exp2 : t2
+          // ----------------------
+          // exp1 ;; epx2 : t2
+          for (
+            texp2 <- visitExp(exp2);
```

Comment:
What about exp1? Should it not be type checked?
(And what type should it have?)